src/string_radix_sort.jl

```diff
@@ -0,0 +1,353 @@
+    n = sizeof(s)
+    if n < skipbytes
+        res = zero(T)
+    elseif n - skipbytes >= sizeof(T)
+        res = ntoh(unsafe_load(Ptr{T}(pointer(s, skipbytes+1))))
+    else
+        ptrs  = pointer(s) + skipbytes
+        remaining_bytes_to_load = n - skipbytes
+        res = load_bits_with_padding(T, ptrs, remaining_bytes_to_load)
+    end
+    return res
+end
+
+# Radix sort for strings
+function sort!(svec::AbstractVector{String}, ::StringRadixSortAlg, o::Perm)
+    sort!(svec, 1, length(svec), StringRadixSort, o)
+end
+
+function sort!(svec::AbstractVector, lo::Int, hi::Int, ::StringRadixSortAlg, o::O) where O <: Union{ForwardOrdering, ReverseOrdering, Perm}
+    if isa(o, Perm)
+        if eltype(o.data) != String
+            throw(ArgumentError("Cannot use StringRadixSort on type $(eltype(o.data))"))
+        end
+        return sortperm_radixsort(o.data, order = o.order)
+    else
+        sort!(svec, lo, hi, StringRadixSort, o)
+    end
+end
+
+function sort!(svec::AbstractVector{String}, lo::Int, hi::Int, ::StringRadixSortAlg, o::O) where O <: Union{ForwardOrdering, ReverseOrdering}
+    if lo >= hi;  return svec;  end
+    # the length subarray to sort
+    l = hi - lo + 1
+
+    # find the maximum string length    
+    lens = maximum(sizeof, svec)
+    skipbytes = lens
+    if lens > 0
+        # while lens > 8
+        #     skipbytes = max(0, skipbytes - 16)
+        #     bits128 = zeros(UInt128, l)
+        #     if o == Reverse
+        #         bits128[lo:hi] .= .~load_bits.(UInt128, @view(svec[lo:hi]), skipbytes)
+        #         sorttwo!(bits128, svec, lo, hi)
+        #     else
+        #         bits128[lo:hi] .= load_bits.(UInt128, @view(svec[lo:hi]), skipbytes)
+        #         sorttwo!(bits128, svec, lo, hi)
+        #     end
+        #     lens -= 16
+        # end
+        while lens > 4
+            skipbytes = max(0, skipbytes - 8)
+            bits64 = zeros(UInt64, l)
+            if o == Reverse
+                bits64[lo:hi] .= .~load_bits.(UInt64, @view(svec[lo:hi]), skipbytes)
+                sorttwo!(bits64, svec, lo, hi)
+            else
+                bits64[lo:hi] .= load_bits.(UInt64, @view(svec[lo:hi]), skipbytes)
+                sorttwo!(bits64, svec, lo, hi)
+            end
+            lens -= 8
+        end
+        if lens > 0
+            skipbytes = max(0, skipbytes - 4)
+            bits32 = zeros(UInt32, l)
+            if o == Reverse
+                bits32[lo:hi] .= .~load_bits.(UInt32, @view(svec[lo:hi]), skipbytes)
+                sorttwo!(bits32, svec, lo, hi)
+            else
+                bits32[lo:hi] .= load_bits.(UInt32, @view(svec[lo:hi]), skipbytes)
+                sorttwo!(bits32, svec, lo, hi)
+            end
+            lens -= 4
+        end
+    end
+    svec
+end
+
+"""
+    sorttwo!(vs, index)
+
+Sort both the `vs` and reorder `index` at the same. This allows for faster sortperm
```

Comment:
This sounds similar to what I experimented at https://github.com/JuliaCollections/SortingAlgorithms.jl/pull/33.